src/object/cli_obj.c

```diff
@@ -5701,14 +5696,55 @@ daos_dc_obj2id(void *ptr, daos_obj_id_t *id)
 	*id = obj->cob_md.omd_id;
 }
 
+/** Disable backward compat code */
+#undef daos_obj_generate_oid
 int
 daos_obj_generate_oid(daos_handle_t coh, daos_obj_id_t *oid,
-		      daos_ofeat_t ofeats, daos_oclass_id_t cid,
+		      daos_otype_t in, daos_oclass_id_t cid,
+		      daos_oclass_hints_t hints, uint32_t args)
+{
+	daos_ofeat_t	feat = (daos_ofeat_t)in;
+	daos_otype_t	type;
+
+	if (feat == DAOS_OF_DKEY_UINT64 | DAOS_OF_KV_FLAT | DAOS_OF_ARRAY)
+		type = DAOS_OT_ARRAY;
+	else if (feat == DAOS_OF_DKEY_UINT64 | DAOS_OF_KV_FLAT |
+			 DAOS_OF_ARRAY_BYTE)
+		type = DAOS_OT_ARRAY_BYTE;
```

Comment:
oops, we miss a:

else if (feat == DAOS_OF_DKEY_UINT64 | DAOS_OF_KV_FLAT)
 		type = DAOS_OT_ARRAY_ATTR;